aide_design/cdc_functions.py

```diff
@@ -121,114 +135,120 @@ def _flow_chem_stock(FlowPlant, ConcDoseMax, ConcStock):
-   
-    index = i_cdc(FlowPlant, ConcDoseMax, ConcStock, 
-                DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, 
-                ENCoag, MinorLossCDCTube)
-    
-    len_cdc_tube = (_length_cdc_tube_array(FlowPlant, ConcDoseMax, ConcStock, 
-                                        DiamTubeAvail, HeadlossCDC, ENCoag, 
-                                        MinorLossCDCTube))[index].magnitude
+                 DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, temp, 
+                 en_chem, KMinor):
+   """The length of tubing may be longer than the max specified if the stock 
+   concentration is too high to give a viable solution with the specified 
+   length of tubing."""
+   index = i_cdc(FlowPlant, ConcDoseMax, ConcStock, 
+                DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, temp,
+                en_chem, KMinor)
+   len_cdc_tube = (_length_cdc_tube_array(FlowPlant, ConcDoseMax, ConcStock, 
+                                        DiamTubeAvail, HeadlossCDC, temp, en_chem, 
+                                        KMinor))[index].magnitude
    
-    return len_cdc_tube
+   return len_cdc_tube
 
 
-@u.wraps(u.m, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m, u.m, None, None], False)
+@u.wraps(u.m, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m, u.m, u.degK, None, None], False)
 def diam_cdc_tube(FlowPlant, ConcDoseMax, ConcStock, 
                   DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, 
-                  ENCoag, MinorLossCDCTube):
+                  temp, en_chem, KMinor):
      
     index = i_cdc(FlowPlant, ConcDoseMax, ConcStock, 
                    DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, 
-                   ENCoag, MinorLossCDCTube)
+                   temp, en_chem, KMinor)
     
     diam_cdc_tube = DiamTubeAvail[index]
      
     return diam_cdc_tube
  
 
-@u.wraps(None, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m, u.m, None, None], False)    
+@u.wraps(None, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m, u.m, u.degK, None, None], False)    
 def n_cdc_tube(FlowPlant, ConcDoseMax, ConcStock, 
           DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, 
-          ENCoag, MinorLossCDCTube):
+          temp, en_chem, KMinor):
     
     index = i_cdc(FlowPlant, ConcDoseMax, ConcStock, 
                   DiamTubeAvail, HeadlossCDC, LenCDCTubeMax, 
-                  ENCoag, MinorLossCDCTube)
+                  temp, en_chem, KMinor)
     
     n_cdc_tube = _n_tube_array(FlowPlant, ConcDoseMax, ConcStock, 
-                  DiamTubeAvail, HeadlossCDC)[index]
+                  DiamTubeAvail, HeadlossCDC, Ratio_Error, KMinor)[index]
     
     return n_cdc_tube
 
 
 
 
 # testing
-FLOW_PLANT = 0.01*u.m**3/u.s
-DIAM_ENGLISH_TUBE_AVAIL = np.array(np.arange(1/16,6/16,1/16))*u.inch
-NU_BLEACH = 1*(u.mm**2/u.s)
-HEADLOSS_MAX = 10*(u.cm)
-CONC_BLEACH_CL2 = 51.4*(u.gram/u.L)
-CONC_CL2 = 2*(u.mg/u.L)                                    
-LENGTH_CDC_TUBE_MAX = 2 * u.m
-EN_CHEM = 2
-K_MINOR = 2
+FlowPlant = 100*u.L/u.s
```

Comment:
Ugh, all of this should be deleted and moved to a test_cdc_functions unittest in tests/